Assets/MixedRealityToolkit-Tests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -0,0 +1,543 @@
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.IsEmpty(MixedRealityToolkit.RegisteredMixedRealityServices);
+            LogAssert.Expect(LogType.Error, $"Unable to find {typeof(ITestExtensionService1).Name} service.");
+        }
+
+        [Test]
+        public void Test_05_02_02_UnregisterMixedRealityExtensionServiceByTypeAndName()
+        {
+            // Redo register test service for a valid instance.
+            Test_05_01_RegisterMixedRealityExtensionService();
+
+            // Retrieve service
+            var extensionService1 = MixedRealityToolkit.Instance.GetService(typeof(ITestExtensionService1));
+
+            // Validate
+            Assert.IsNotNull(extensionService1);
+
+            var success = MixedRealityToolkit.UnregisterService(typeof(ITestExtensionService1), extensionService1.Name);
+
+            // Validate non-existent service
+            var isServiceRegistered = MixedRealityToolkit.Instance.IsServiceRegistered<ITestExtensionService1>();
+
+            // Tests
+            Assert.IsTrue(success);
+            Assert.IsFalse(isServiceRegistered);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.IsEmpty(MixedRealityToolkit.RegisteredMixedRealityServices);
+            LogAssert.Expect(LogType.Error, $"Unable to find {typeof(ITestExtensionService1).Name} service.");
+        }
+
+        [Test]
+        public void Test_05_03_RegisterMixedRealityExtensionServices()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Add test ExtensionService
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestExtensionService1), new TestExtensionService1("Test ExtensionService 1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestExtensionService2), new TestExtensionService2("Test ExtensionService 2", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestFailService), new TestFailService());
+            LogAssert.Expect(LogType.Error, $"{typeof(ITestFailService).Name} does not implement {typeof(IMixedRealityService).Name}.");
+
+            // Retrieve all registered IMixedRealityExtensionServices
+            var extensionServices = MixedRealityToolkit.Instance.GetActiveServices(typeof(IMixedRealityExtensionService));
+
+            // Tests
+            Assert.IsNotNull(MixedRealityToolkit.Instance.ActiveProfile);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.AreEqual(2, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+            Assert.AreEqual(extensionServices.Count, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+        }
+
+        [Test]
+        public void Test_05_04_UnregisterMixedRealityExtensionServicesByType()
+        {
+            // Redo register test services for a valid instances.
+            Test_05_03_RegisterMixedRealityExtensionServices();
```

Comment:
Rather than re-running a test (which may fail) please consider a helper function that sets the desired initial conditions?